homeassistant/components/http/__init__.py

```diff
@@ -195,13 +199,28 @@ def __init__(self, hass, development, api_password, ssl_certificate,
         self.app[KEY_LOGIN_THRESHOLD] = login_threshold
         self.app[KEY_DEVELOPMENT] = development
 
+        try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.bind((server_host, server_port))
+            if server_port == 0:
```

Comment:
I'll switch to use this helper instead. It should allow us to roll back all the changes outside of the test system.

The drawback to this approach is that it introduces a race where the port could be reserved in-between checking for an available port number, and later binding to it. However, since we're only using this in our test system it shouldn't be too big of a deal.